#! /bin/sh
patch -p1 -f $* < $0
exit $?

For the logo, see actually 08-w32-logo.patch.

2006-07-14  Marcus Brinkmann <mb@g10code.com>

	* w32-resource.rc: New file.
	* w32-logo.ico: New file.
	* src/Makefile.am: Define rules to compile resource files.
	(sylpheed_claws_LDFLAGS): Add $(sylpheed_claws_res_ldflag).
	(sylpheed_claws_DEPENDENCIES): Add $(sylpheed_claws_deps).
	(EXTRA_DIST): Add w32-logo.ico and w32-resource.rc.


diff -rupN sylpheed-claws-2.3.1-orig/src/Makefile.am sylpheed-claws-2.3.1/src/Makefile.am
--- sylpheed-claws-2.3.1-orig/src/Makefile.am	2006-06-20 10:31:43.000000000 +0200
+++ sylpheed-claws-2.3.1/src/Makefile.am	2006-07-14 01:33:38.000000000 +0200
@@ -11,6 +11,21 @@ SUBDIRS = common gtk $(etpan_dir) . plug
 
 bin_PROGRAMS = sylpheed-claws
 
+if PLATFORM_WIN32
+LTRCCOMPILE = $(LIBTOOL) --mode=compile $(RC) \
+     `echo $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) | \
+     sed -e 's/-I/--include-dir /g;s/-D/--define /g'`
+
+%.o : %.rc
+	$(LTRCCOMPILE) -i $< -o $@
+
+sylpheed_claws_res_ldflag = -Wl,w32-resource.o
+sylpheed_claws_deps = w32-resource.o
+else
+sylpheed_claws_res_ldflag =
+sylpheed_claws_deps =
+endif
+
 sylpheed_claws_SOURCES = \
 	account.c \
 	action.c \
@@ -288,6 +303,7 @@ BUILT_SOURCES = \
 	quote_fmt_parse.h
 
 EXTRA_DIST = \
+	w32-logo.ico w32-resource.rc \
 	pixmaps/address_book.xpm \
 	pixmaps/address_search.xpm \
 	pixmaps/address.xpm \
@@ -410,7 +426,11 @@ arch_ldflags :=
 endif
 
 sylpheed_claws_LDFLAGS = \
-	$(arch_ldflags) -export-dynamic
+	$(arch_ldflags) -export-dynamic $(sylpheed_claws_res_ldflag)
+
+sylpheed_claws_DEPENDENCIES = $(sylpheed_claws_deps) \
+	$(etpan_library) \
+	gtk/libsylpheedgtk.la
 
 sylpheed_claws_LDADD = \
 	$(etpan_library) \
diff -rupN sylpheed-claws-2.3.1-orig/src/w32-resource.rc sylpheed-claws-2.3.1/src/w32-resource.rc
--- sylpheed-claws-2.3.1-orig/src/w32-resource.rc	1970-01-01 01:00:00.000000000 +0100
+++ sylpheed-claws-2.3.1/src/w32-resource.rc	2006-07-14 01:31:03.000000000 +0200
@@ -0,0 +1 @@
+1 ICON "w32-logo.ico"
